wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+SecurityLevel.swift

```diff
@@ -728,6 +728,42 @@ class ZMConversationTests_SecurityLevel: ZMConversationTestsBase {
         }
     }
 
+    func testItMarksMLSConversationAsNotVerifiedAfterResendMessage() async {
+        // GIVEN
+        let conversation = await syncMOC.perform {
+            let conversation = ZMConversation.insertNewObject(in: self.syncMOC)
+            conversation.conversationType = .group
+            conversation.messageProtocol = .mls
+            conversation.mlsVerificationStatus = .degraded
+
+            return conversation
+        }
+
+        let message = await syncMOC.perform {
+            return try! conversation.appendText(content: "foo") as! ZMOTRMessage
+        }
+
+        await syncMOC.perform {
+            XCTAssertEqual(message.deliveryState, .failedToSend)
+            XCTAssertTrue(message.isExpired)
+            XCTAssertTrue(message.causedSecurityLevelDegradation)
+        }
+
+        // WHEN
+        await uiMOC.perform {
+            let uiConversation = try? self.uiMOC.existingObject(with: conversation.objectID) as? ZMConversation
+            uiConversation?.acknowledgePrivacyWarning(withResendIntent: true)
+        }
```

Comment:
here you could also do the unwrapping similarly as above if you're expecting the conversation to exist

Reply:
done